packages/shared/lib/ledger.ts

```diff
@@ -38,16 +43,38 @@ export function calculateLedgerDeviceState(status: LedgerStatus): LedgerDeviceSt
     if (locked) {
         return LedgerDeviceState.Locked
     } else {
-        if (app?.name === AppName.IOTA) {
-            return LedgerDeviceState.Connected
-        } else if (app?.name === AppName.IOTALegacy) {
-            return LedgerDeviceState.LegacyConnected
-        } else {
-            return connected ? LedgerDeviceState.AppNotOpen : LedgerDeviceState.NotDetected
+        switch(app?.name) {
+            default:
+                if(connected) {
+                    /**
+                     * NOTE: "BOLOS" is the name of the Ledger operating system and is
+                     * sometimes registered as an app.
+                     */
+                    return (app?.name && app?.name !== 'BOLOS') ? LedgerDeviceState.OtherConnected : LedgerDeviceState.AppNotOpen
+                } else {
+                    return LedgerDeviceState.NotDetected
+                }
+            case LedgerAppName.IOTA:
+                return LedgerDeviceState.Connected
+            case LedgerAppName.IOTALegacy:
+                return LedgerDeviceState.LegacyConnected
         }
     }
 }
 
+export function getLedgerOpenedAppName(): Promise<LedgerApp> {
+    return new Promise<LedgerApp>((resolve, reject) => {
+        api.getLedgerDeviceStatus(ledgerSimulator, {
+            onSuccess(response: Event<LedgerStatus>) {
+                resolve(response.payload?.app)
```

Comment:
Is it possible that `payload` doesn't have `app` prop?

Reply:
It is possible, and the result is `null` in that case. I originally needed the name from this data, but I changed the function name to make more sense even if it returns null (just removed "name" so `getLedgerOpenedApp(...)`.